### PR TITLE
[url_launcher_linux] add file scheme support for canLaunch

### PR DESCRIPTION
**Pull request by jokerttu**
Fixes `canLaunch` implementation for `file` scheme. 
`canLaunch` incorrectly returned a `false` response to launchable file scheme URLs.
Fix uses `g_content_type_guess` to guess `content_type` from the filepath and `g_app_info_get_default_for_type` to detect if any application is associated for found `content_type`.

*List which issues are fixed by this PR. You must list at least one issue.*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

**Comment by jokerttu** [comment]
Plugin template for linux platform will be updated later, improving unit test support a bit, and tests need to be updated.

**Comment by jokerttu** [comment]
Even when example update is related to file support (to provide example with it) it might be that stripping ot off from this commit, like as well not commit related function name change would be good to keep commit as atomic as possible... but then again creating own issues for those might increase work unnecessarily.

File: packages/url_launcher/url_launcher_linux/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.0.3
+
+* Fix canLaunch detection for `file` scheme
```
Comment (tomgilder) [inline packages/url_launcher/url_launcher_linux/CHANGELOG.md:3]: Probably worth saying this is just on Linux

Comment (jokerttu) [reply packages/url_launcher/url_launcher_linux/CHANGELOG.md:3]: Hmm, maybe, this is url_launcher_linux package so I not sure does it need that information

**Pull request closed without merging**